### PR TITLE
Add end times for Session post type

### PR DESCRIPTION
**Pull request by iandunn**
Fixes https://meta.trac.wordpress.org/ticket/3623

Will be needed for the (full) Schedule block (see https://meta.trac.wordpress.org/ticket/3117), and the Live Schedule block.

I've tested it with the full Schedule block and it works well. @ryelle , can you test it w/ the Live Schedule block, and make sure it meets your needs for that? It adds new fields to the `Session Info` metabox.

### CODE REVIEW

File: public_html/wp-content/plugins/wc-post-types/inc/rest-api.php
Comment (ryelle) [inline public_html/wp-content/plugins/wc-post-types/inc/rest-api.php]: Could this be saved (or just exposed) as one value, `_wcpt_session_length`? This way we can avoid repeated calculations, since we'll almost always want to know the full duration, not just hours/minutes.

File: public_html/wp-content/plugins/wc-post-types/wc-post-types.php
Comment (ryelle) [inline public_html/wp-content/plugins/wc-post-types/wc-post-types.php]: It looks like while two labels for the same input is technically valid html, [screen readers might not read this correctly.](https://stackoverflow.com/questions/2829936/can-an-input-field-have-two-labels/3992160#3992160) A more standard approach would be a `fieldset` to wrap both inputs, and `legend` for the first label; and then you don't need the `aria-label`s either (the legend will set the context for both selects, then read their labels).

File: public_html/wp-content/plugins/wc-post-types/wc-post-types.php
Comment (ryelle) [inline public_html/wp-content/plugins/wc-post-types/wc-post-types.php]: Same comment as on the API side - I think it would make more sense to calculate the duration (in seconds, probably) and save that in one single field.

**Review by ryelle — COMMENTED** [review]
I left feedback on this as-is, but the new UI control could use some design feedback too.

@melchoyce we're talking about just the "Length" fields in the screenshots here:

Artboard

**Comment by melchoyce** [comment]
Is this in the sidebar? Hard to tell but looks like maybe no?

Would be great to simplify and streamline these fields — maybe something like:

image

Instead of dropdowns, it seems like it would be easier for folks to enter numbers directly into a number input field. Then they could also use arrow keys to adjust increments.

Ideally, we'd also use the calendar/datepicker component in addition to the field, like the publish field:

image

One of the things I struggle with when entering sessions is, well, dates — numbers tend to get jumbled in my head. Having a visual calendar for scheduling vastly improves the overall experience for me.

**Comment by ryelle** [comment]
> Ideally, we'd also use the calendar/datepicker component in addition to the field, like the publish field

yeah, I'd love to get this using gutenberg stuff instead of a plain old metabox, but i think that's an issue for another time 🙂

**Comment by melchoyce** [comment]
Ah, it's a metabox. Fair enough.

File: public_html/wp-content/plugins/wc-post-types/inc/rest-api.php
Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/inc/rest-api.php]: 🤔 IIRC there was a reason I kept them separate, but maybe that was just dictated by the UI being two separate fields. I'll see if I can remember.

**Comment by iandunn** [comment]
Added the `fieldset` and `number` fields in d9bdd2acd053c97a58c4de834ef29ad117254f4f, but I haven't tested how well that works w/ the full Schedule block yet.

I'll wait to see what happens with #303 before doing anything else, to avoid spending time on things that would need to be refactored.

**Comment by ryelle** [comment]
@iandunn What is this blocked by?

**Comment by iandunn** [comment]
Did you notice the comment above? Let me know if there's anything missing that you need.

**Comment by ryelle** [comment]
Yes, but it's not clear from the comment what's blocking. #303 is just an experiment, no one's even replied to it yet— when this merges I'll just update it.

**Comment by iandunn** [comment]
If #303 can be set to `blocked` until this is merged, then I'm happy to pick this up again. 

Otherwise I'd prefer to wait until #303 is merged before restarting this, to avoid the potential for having to redo the UI.

**Comment by ryelle** [comment]
#303 does not block this though— there's no timeline on merging it, and this PR is more important. The UI in this PR is fine, the only change that still needs to happen is merging the fields into one "length" field.

**Comment by iandunn** [comment]
I think we just have different opinions, but it seems like you're fine with waiting to merge #303 until this is done, so I'll go ahead and remove the `blocked` label here and add it to #303. Feel free to reverse that if I'm misunderstand your perspective.

File: public_html/wp-content/plugins/wc-post-types/wc-post-types.php
Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/wc-post-types.php]: That's a great catch, thanks! Fixed in d9bdd2acd053c97a58c4de834ef29ad117254f4f

File: public_html/wp-content/plugins/wc-post-types/inc/rest-api.php
Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/inc/rest-api.php]: I think it was just a result of the UI. I combined them in 4aa7acb286d75469cc05253ee976fb5f38ed05f4, and that seems to work fine for the purposes of the Schedule block.

File: public_html/wp-content/plugins/wc-post-types/wc-post-types.php
Comment (iandunn) [reply public_html/wp-content/plugins/wc-post-types/wc-post-types.php]: Done in 4aa7acb286d75469cc05253ee976fb5f38ed05f4

**Comment by iandunn** [comment]
I think this is ready to merge, but could you test it w/ the changes you want to make to the Live Schedule block, to make sure it works for that as well?

**Comment by iandunn** [comment]
[ The merge conflict is just b/c of cherry picking the coding standards commit to production, I'll rebase ]

**Review by ryelle — APPROVED** [review]
Looks good! I haven't built anything for the Live Schedule yet, but this matches what I was thinking.

I see there are still some failing phpcs checks, were you going to fix those too, or leave as-is? (not a blocker, just unclear from your comment above)

**Pull request merged**
